superset/common/query_context.py

```diff
@@ -152,13 +152,39 @@ def get_single_payload(
         self, query_obj: QueryObject, force_cached: Optional[bool] = False,
     ) -> Dict[str, Any]:
         """Return results payload for a single quey"""
-        if self.result_type == utils.ChartDataResultType.QUERY:
+        result_type = query_obj.result_type or self.result_type
+        datasource = query_obj.datasource or self.datasource
+        if result_type == utils.ChartDataResultType.COLUMNS:
```

Comment:
since these can grow further, could make the code look cleaner if we create a simple `Dict` that maps to functions with the actual comprehensions, something like:

``` python
result_type_map = {
    utils.ChartDataResultType.COLUMNS: get_result_type_columns,
    utils.ChartDataResultType.TIMEGRAINS: get_result_type_time_grains,
    utils.ChartDataResultType.QUERY: get_result_type_query
}
```

Reply:
This is a superb idea!